data/preserve_data.rb

```diff
@@ -0,0 +1,20 @@
+require 'json'
+
+def save_book(data, filename)
+  data_array = data.map(&:to_h)
+  File.write(filename, JSON.generate(data_array))
+end
+
+def save_label(_data, filename)
+  data_array = ddata.map(&:to_h)
```

Comment:
I'm getting an error when trying to create a new label. Do you think ddata.map should be _data.map here?

Reply:
done